homeassistant/components/orangepi_gpio/binary_sensor.py

```diff
@@ -0,0 +1,68 @@
+"""Support for binary sensor using Orange Pi GPIO."""
+import logging
+
+from homeassistant.components import orangepi_gpio
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, PLATFORM_SCHEMA)
+from homeassistant.const import DEVICE_DEFAULT_NAME
+
+from . import CONF_PINMODE
+from .const import CONF_INVERT_LOGIC, CONF_PORTS, PORT_SCHEMA
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(PORT_SCHEMA)
```

Comment:
Since we know that more platforms are likely incoming, it would be better to have all config under the integrating component section, and set up platforms via discovery helper. See the netgear_lte component for example.